app/executor.php

```diff
@@ -0,0 +1,1372 @@
+    });
+
+    if ($tag->getAttribute('functionId') !== $function->getId()) {
+        throw new Exception('Tag not found', 404);
+    }
+
+    // Grab execution document if exists
+    // It it doesn't exist, create a new one.
+    $execution = Authorization::skip(function () use ($database, $executionId, $userId, $function, $tag, $trigger, $functionId) {
+        return (!empty($executionId)) ? $database->getDocument('executions', $executionId) : $database->createDocument('executions', new Document([
+            '$id' => $executionId,
+            '$read' => (!$userId == '') ? ['user:' . $userId] : [],
+            '$write' => [],
+            'dateCreated' => time(),
+            'functionId' => $function->getId(),
+            'tagId' => $tag->getId(),
+            'trigger' => $trigger, // http / schedule / event
+            'status' => 'processing', // waiting / processing / completed / failed
+            'statusCode' => 0,
+            'stdout' => '',
+            'stderr' => '',
+            'time' => 0.0,
+            'search' => implode(' ', [$functionId, $executionId]),
+        ]));
+    });
+
+    if (false === $execution || ($execution instanceof Document && $execution->isEmpty())) {
+        throw new Exception('Failed to create or read execution');
+    }
+
+
+    if ($build->getAttribute('status') == 'building') {
+        Console::error('Execution Failed. Reason: Code was still being built.');
+
+        $execution->setAttribute('status', 'failed')
+            ->setAttribute('statusCode', 500)
+            ->setAttribute('stderr', 'Tag is still being built.')
+            ->setAttribute('time', 0);
+
+        Authorization::skip(function () use ($database, $execution) {
+            return $database->updateDocument('executions', $execution->getId(), $execution);
+        });
+        throw new Exception('Tag is still being built.');
+    }
+
+    // Check if runtime is active
+    $runtime = (isset($runtimes[$function->getAttribute('runtime', '')]))
+        ? $runtimes[$function->getAttribute('runtime', '')]
+        : null;
+
+    if (\is_null($runtime)) {
+        throw new Exception('Runtime "' . $function->getAttribute('runtime', '') . '" is not supported');
+    }
+
+    // Process environment variables
+    $vars = \array_merge($function->getAttribute('vars', []), [
+        'APPWRITE_FUNCTION_ID' => $function->getId(),
+        'APPWRITE_FUNCTION_NAME' => $function->getAttribute('name', ''),
+        'APPWRITE_FUNCTION_TAG' => $tag->getId(),
+        'APPWRITE_FUNCTION_TRIGGER' => $trigger,
+        'APPWRITE_FUNCTION_RUNTIME_NAME' => $runtime['name'],
+        'APPWRITE_FUNCTION_RUNTIME_VERSION' => $runtime['version'],
+        'APPWRITE_FUNCTION_EVENT' => $event,
+        'APPWRITE_FUNCTION_EVENT_DATA' => $eventData,
+        'APPWRITE_FUNCTION_DATA' => $data,
+        'APPWRITE_FUNCTION_USER_ID' => $userId,
+        'APPWRITE_FUNCTION_JWT' => $jwt,
+        'APPWRITE_FUNCTION_PROJECT_ID' => $projectId,
+    ]);
```

Comment:
Would it be possible to include an endpoint (main from .ENV probably) and some temporary API key (900s)? That would be awesome, so you don't need to set anything manually when trying to connect to Appwrite SDK.